controllers/glance_common.go

```diff
@@ -19,13 +19,31 @@ package controllers
 import (
 	"context"
 	"fmt"
+
 	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/secret"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// fields to index to reconcile when change
+const (
+	passwordSecretField     = ".spec.secret"
+	caBundleSecretNameField = ".spec.tls.caBundleSecretName"
+	tlsAPIInternalField     = ".spec.tls.api.internal"
+	tlsAPIPublicField       = ".spec.tls.api.public"
```

Comment:
Why are the first two indexed on the `String` secret name (`Secret` and `CaBundleSecretName`) but the last two are indexing on the `GenericService` field (instead of its embedded `SecretName` `String` field)?

Reply:
its a mistake, thanks for the pointer, it should be `.spec.tls.api.internal.secretName`